TrafficReplayer/pull_then_poll_cw_logs.py

```diff
@@ -0,0 +1,92 @@
+    current_events_raw = current_response["events"]
+    messages.extend([event["message"] for event in current_events_raw])
+    current_token = current_response["nextForwardToken"]
+
+    next_response = logs_client.get_log_events(logGroupName=DEFAULT_LOG_GROUP,
+                                               logStreamName=DEFAULT_LOG_STREAM,
+                                               startFromHead=True, nextToken=current_token)
+    next_token = next_response["nextForwardToken"]
+
+    while current_token != next_token:
+        print("More events to pull...")
+        current_response = next_response
+        current_events_raw = current_response["events"]
+        messages.extend([event["message"] for event in current_events_raw])
+        current_token = current_response["nextForwardToken"]
+
+        next_response = logs_client.get_log_events(logGroupName=DEFAULT_LOG_GROUP,
+                                                   logStreamName=DEFAULT_LOG_STREAM,
+                                                   startFromHead=True, nextToken=current_token)
+        next_token = next_response["nextForwardToken"]
+
+    print(f"Pulled {len(messages)} events")
+
+    print(f"Writing events to file: {DEFAULT_FILE}")
+    with open(DEFAULT_FILE, 'w') as output_file:
+        output_file.write("\n".join(messages))
+        output_file.write("\n")
+
+    # Now that all currently available CloudWatch events were logged.
+    # We start checking for new events every now and then, and append new ones to already existing file, if available
+
+    while True:
+        messages.clear()
+
+        next_response = logs_client.get_log_events(logGroupName=DEFAULT_LOG_GROUP,
+                                                   logStreamName=DEFAULT_LOG_STREAM,
+                                                   startFromHead=False, nextToken=current_token)
+        next_token = next_response["nextForwardToken"]
+
+        if current_token != next_token:
+            current_response = next_response
+            current_events_raw = current_response["events"]
+            messages.extend([event["message"] for event in current_events_raw])
+            current_token = current_response["nextForwardToken"]
+
+            next_response = logs_client.get_log_events(logGroupName=DEFAULT_LOG_GROUP,
+                                                       logStreamName=DEFAULT_LOG_STREAM,
+                                                       startFromHead=False, nextToken=current_token)
+            next_token = next_response["nextForwardToken"]
+
+            if len(messages):
+                print(f"Found new log events. Writing events to file: {DEFAULT_FILE} - Checking again in 10 seconds")
+                with open(DEFAULT_FILE, 'a') as output_file:
+                    output_file.write("\n".join(messages))
+
+        if not len(messages):
+            print("No new log events found, sleeping for 10 seconds then checking again")
+
+        time.sleep(10)
```

Comment:
Why do you need to sleep?  Does the API not allow for [long-polling](https://javascript.info/long-polling)?

Reply:
My thought was that we only need to wake this up every now and then, that's why I semi-randomly picked 10 seconds.
I'm not sure if it has long-polling, but that sounds like a good idea! I'll look into it.

Reply:
I didn't find any long-polling available for CW